### PR TITLE
Support SDL2 if available on the system

### LINKED ISSUE #1234: Porting SDL_Joypad device to SDL2
We should port the joypad driver to SDL2 since SDL1 is not supported on vcpkg, I made a few tests and the only calls to replace seem to be  `SDL_JoystickName` -> `SDL_JoystickNameForIndex`.

Tested, it works, but I had to tweak `FindSDL.cmake` in CMake 3.7 to have it find the SDL2 library

### PR DESCRIPTION
**Pull request by drdanz**
Fixes #1234

I have no idea how to test the driver, and I don't have a controller, can anyone test if this works?
CC @randaz81, @aerydna

### CODE REVIEW

**Comment by drdanz** [comment]
Ok, I managed to test this with an old PS1 controller + USB adaptor... It works for me

File: src/devices/SDLJoypad/CMakeLists.txt [hunk 1]
```diff
@@ -24,9 +24,15 @@ if(ENABLE_SDLJoypad)
                                                       YARP_sig
                                                       YARP_dev)
 
-  target_include_directories(yarp_SDLJoypad SYSTEM PRIVATE ${SDL_INCLUDE_DIR})
-  target_link_libraries(yarp_SDLJoypad PRIVATE ${SDL_LIBRARY})
-#   list(APPEND YARP_${YARP_PLUGIN_MASTER}_PRIVATE_DEPS SDL) (not using targets)
+  if(YARP_HAS_SDL2)
+    target_include_directories(yarp_SDLJoypad SYSTEM PRIVATE ${SDL2_INCLUDE_DIRS})
+    target_link_libraries(yarp_SDLJoypad PRIVATE ${SDL2_LIBRARIES})
```
Comment (randaz81) [inline src/devices/SDLJoypad/CMakeLists.txt]: On my Windows machine, `${SDL2_LIBRARIES}` seems not be available.
I had to replace it with `SDL2::SDL2 SDL2::SDL2main` as suggested by vcpkg:

```
$ ./vcpkg.exe install sdl2:x64-windows
Computing installation plan...
The following packages are already installed:
    sdl2[core]:x64-windows
Package sdl2:x64-windows is already installed

Total elapsed time: 13.86 us

The package sdl2:x64-windows provides CMake targets:

    find_package(SDL2 CONFIG REQUIRED)
    target_link_libraries(main PRIVATE SDL2::SDL2 SDL2::SDL2main)
```

File: src/devices/SDLJoypad/CMakeLists.txt [hunk 2]
```diff
@@ -24,9 +24,15 @@ if(ENABLE_SDLJoypad)
                                                       YARP_sig
                                                       YARP_dev)
 
-  target_include_directories(yarp_SDLJoypad SYSTEM PRIVATE ${SDL_INCLUDE_DIR})
-  target_link_libraries(yarp_SDLJoypad PRIVATE ${SDL_LIBRARY})
-#   list(APPEND YARP_${YARP_PLUGIN_MASTER}_PRIVATE_DEPS SDL) (not using targets)
+  if(YARP_HAS_SDL2)
+    target_include_directories(yarp_SDLJoypad SYSTEM PRIVATE ${SDL2_INCLUDE_DIRS})
```
Comment (randaz81) [inline src/devices/SDLJoypad/CMakeLists.txt]: `${SDL2_INCLUDE_DIRS}` does not exists on windows?
see next comment.

File: src/devices/SDLJoypad/CMakeLists.txt (hunk 1 above)
Comment (traversaro) [reply src/devices/SDLJoypad/CMakeLists.txt]: Unfortunatly I checked SDL build system, and apparently the exported CMake  variable depend on wheter SDL has been build with autotools (as I imagine the Debian package is built) or with CMake (as done in vcpkg), so it is a big mess...

**Pull request closed without merging**